test/regression/issue/decorator-metadata-unions.test.ts

```diff
@@ -0,0 +1,367 @@
+        };
+      }
+
+      class User {
+        @Property()
+        profile: Profile | null = null;
+      }
+
+      class Profile {
+        @Property()
+        user: User | null = null;
+      }
+
+      console.log("SUCCESS");
+    `,
+    "package.json": JSON.stringify({
+      dependencies: {
+        "reflect-metadata": "latest",
+      },
+    }),
+  });
+
+  // Install dependencies
+  const installProc = Bun.spawn({
+    cmd: [bunExe(), "install"],
+    cwd: String(dir),
+    env: bunEnv,
+    stdout: "pipe",
+    stderr: "pipe",
+  });
+  await installProc.exited;
+
+  await using proc = Bun.spawn({
+    cmd: [bunExe(), "index.ts"],
+    env: bunEnv,
+    cwd: String(dir),
+    stderr: "pipe",
+    stdout: "pipe",
+  });
+
+  const [stdout, stderr, exitCode] = await Promise.all([proc.stdout.text(), proc.stderr.text(), proc.exited]);
+
+  // Should not have TDZ errors
+  expect(stderr).not.toContain("Cannot access");
+  expect(stderr).not.toContain("before initialization");
+
+  // Should emit Object for union types (treating as strictNullChecks: true)
+  expect(stdout).toContain("profile: Object");
+  expect(stdout).toContain("user: Object");
+  expect(stdout).toContain("SUCCESS");
+  expect(exitCode).toBe(0);
+}, 30000);
+
+test("Bun matches TypeScript with strictNullChecks", async () => {
+  const node = nodeExe();
+  if (!node) {
+    console.log("Skipping test: Node.js not found");
+    return;
+  }
+
+  using dir = tempDir("tsc-bun-comparison", {
+    "package.json": JSON.stringify({
+      name: "tsc-comparison",
+      devDependencies: {
+        "typescript": "~4.9.0",
+      },
+    }),
+    "tsconfig.json": JSON.stringify({
+      compilerOptions: {
+        experimentalDecorators: true,
+        emitDecoratorMetadata: true,
+        target: "ES2015",
+        module: "commonjs",
+        strict: true,
+        strictNullChecks: true,
+      },
+      files: ["test.ts"],
+    }),
+    "test.ts": `
+      function Property() {
+        return function (_target: any, _propertyKey: string) {};
+      }
+
+      class User {
+        @Property()
+        name: string = "";
+
+        @Property()
+        profile: Profile | null = null;
+
+        @Property()
+        settings: Settings | undefined;
+      }
+
+      class Profile {
+        @Property()
+        bio: string = "";
+      }
+
+      class Settings {
+        @Property()
+        theme: string = "light";
+      }
+    `,
+  });
+
+  // Install TypeScript
+  const installProc = Bun.spawn({
+    cmd: [bunExe(), "install"],
+    cwd: String(dir),
+    env: bunEnv,
+    stdout: "pipe",
+    stderr: "pipe",
+  });
+  await installProc.exited;
+
+  // Compile with TypeScript
+  const tscProc = Bun.spawn({
+    cmd: [bunExe(), "x", "tsc"],
+    cwd: String(dir),
+    env: bunEnv,
+    stdout: "pipe",
+    stderr: "pipe",
+  });
+  await tscProc.exited;
+
+  // Compile with Bun
+  const bunBuildProc = Bun.spawn({
```

Comment:
Can you use Bun.build and then await Promise.all([buildpromise, tscProc.exited])